### ISSUE TITLE
Bugs and Small Improvement Request

### ISSUE THREAD

**Issue by eaganlai** [Lurkars/gloomhavensecretariat#336]
Thank you for making this amazing app! Since you are so active in taking feedback (on Reddit and BGG) I guess I might share some of my suggestion with you! Below are a bunch of nitpicking for you to waste more time in this non-profitable but hopefully fun project. 

**Feedback**
- [Bug] Failed to send feedback. After clicking the send button, nothings happens.

**Setting**
- [Improvement] Setting should be adapted to one column when viewed on mobile.

**UI - General**
- [Bug] On mobile, when Bar Size is set to large, Elements might be pushed out of the screen.
- [Improvement] For the reminder text of the negative/positive scenario effects, make it bigger, sitting in the center of the screen. Insert line break after each characters name (maybe put their icons at the left of the name, coz why not?), and then show the text of the effects below the names in a full sentence. The whole reminder could be easily missed right now, which kinda defeats its purpose, and the name could be clipped if they are too long (or when the characters got two names).
- [Improvement] When setting Initiative, simply put a wide bar button saying "Long Rest" under zero is more intuitive than a 'history' icon.
- [Improvement] For the Loot Deck, when the loot deck drawer is closed, make the loot deck button BIGGER, and remove the 'enlarge' and 'setting button' in this state. When the drawer is open, show the 'enlarge' and 'Setting' buttons above the deck, in a slightly bigger way.
- [Improvement] Similar to how the count of gained XP and looted Coins are displayed originally, we should show the count looted loot deck of each play for more blaming and jealousy drama. Maybe we could replace the looted coin icon in a Frosthaven campaign. If we are not replacing it, we could make them in vertical arrangement, as in we have three column of info: Loot Deck (hidden if zero), XP, Loot Coins, and in each column, the icon will be on the top and the count shall sit below the icon.
- [Improvement] If possible, long press the monster avatar to show the art of their whole standee. Sometimes it could be hard to identify.

**UI - Party Sheet**
- [Improvement] In the Supply section of the Party Sheet, the 'ADD' and 'DEDUCT' button are no way press-able with my fat fingers. Please make them two square buttons side by side. Honestly, redesign the party sheet and make it in style of the HP/EXP/LOOT/BLESS/CURSE management would be perfect.

**UI - Character Sheet**
- [Improvement] For Character Sheet, make it a vertical single column of section starting from name, XP & Gold, Resources, Masteries, Perks, and then Notes. For the Perk section, do not put the check boxes and text in the same row. One row for check boxes and one row for text.
- [Improvement] For the items part of the Character Sheet, put the drop down on the left-most followed by the number input box with an "ADD" button closely tied to the input box. I don't think the left and right arrows are helpful (when potentially navigating hundreds of items sorted in mysteriously numbered order). Adding a 'Available Items' button to open up an item list/index would be a better option.

**Other**
- Thank you for the switching forms function!! It is so amazing and amazingly useless (gameplay-wise), but I loved it so much. I got a funny second name for my Geminate and it shows!!!! It is such attentions to details made me in love with you.

Thank you for this amazing web app!! LOVE YOU!!!!!!!!!! I have tried x-haven but I think GS is much more appealing visually and it is an absolute priority for me! Glad we have so many amazing, humble and talented people willing to pick up the legacy of GHH. I would give you a huge kiss if I can. May you loot all the coins in the loot deck.

**Comment by Lurkars** [comment]
Try to give you short feedback. First of all thanks for taking your time to report in such detail. Since many issues affect mobile experience, I have to ask you if you can provide your device information especially your display resolution, because some stuff sounds to me, that you targeted a very niché spot with your device.

>     * [Bug] Failed to send feedback. After clicking the send button, nothings happens.

The button is a simple `mailto:issue@gloomhaven-secretariat.de` link, so seems that your device do not respond to those links or please check if your email client opened in background or so. I don't think I can change this behaviour, because I won't provide a mail form into the app. Other possibility is, that you wrote a very long text and the browser wasn't able to handle this. So you may also check the functionality with a short message. Anyway, when your device is not able to handle `mailto:` (so sending emails), it's possible that the button just doesn't work.

>     * [Improvement] Setting should be adapted to one column when viewed on mobile.

This is definetly already the case, but the device width is currently set to `500px`, so it's possible, that your device has a slightly bigger width. I will check, if I can add something like "mobile" mode, where you use the whole screen regardlessly of acutal width.

>     * [Bug] On mobile, when Bar Size is set to large, Elements might be pushed out of the screen.

I will optimize sizing and make the space scrollable on overflow.

>     * [Improvement] For the reminder text of the negative/positive scenario effects, make it bigger, sitting in the center of the screen. Insert line break after each characters name (maybe put their icons at the left of the name, coz why not?), and then show the text of the effects below the names in a full sentence. The whole reminder could be easily missed right now, which kinda defeats its purpose, and the name could be clipped if they are too long (or when the characters got two names).

I will take a look.

>     * [Improvement] When setting Initiative, simply put a wide bar button saying "Long Rest" under zero is more intuitive than a 'history' icon.

I am unsure about this, because there are many thinks that may not be intuitive and you just need to learn. If you once knew it, it just looks cleaner and easier then always having a big text button on this view. But we can discuss on that.

>     * [Improvement] For the Loot Deck, when the loot deck drawer is closed, make the loot deck button BIGGER, and remove the 'enlarge' and 'setting button' in this state. When the drawer is open, show the 'enlarge' and 'Setting' buttons above the deck, in a slightly bigger way.

The `fullscreen` and `settings` buttons are just a miss, I already do that for the AM deck, no idea why missed on loot. Will come definitely, also making the icon bigger when closed is a good idea. About enlarging the icons, I have to check what looks good.

>     * [Improvement] Similar to how the count of gained XP and looted Coins are displayed originally, we should show the count looted loot deck of each play for more blaming and jealousy drama. Maybe we could replace the looted coin icon in a Frosthaven campaign. If we are not replacing it, we could make them in vertical arrangement, as in we have three column of info: Loot Deck (hidden if zero), XP, Loot Coins, and in each column, the icon will be on the top and the count shall sit below the icon.

I think adding an indicator about how many loot cards are looted by the char is a first good improvement will do that. On the other stuff I am still unsure. For maintaining reasons I already stated out somewhere that I don't want to change too much stuff between editions, so that's always a trade-off.

>     * [Improvement] If possible, long press the monster avatar to show the art of their whole standee. Sometimes it could be hard to identify.

Interesting, cannot imagine that images are too small to identify, but your request shouldn't be much work either, so maybe just implement it. As stated above, happy about your screen dimensions and zoom level...

>     * [Improvement] In the Supply section of the Party Sheet, the 'ADD' and 'DEDUCT' button are no way press-able with my fat fingers. Please make them two square buttons side by side. Honestly, redesign the party sheet and make it in style of the HP/EXP/LOOT/BLESS/CURSE management would be perfect.

You mean the buttons of the input fields? That is build in from browser, so honestly I am unsure if I will touch those. Also it should be numeric input, so you can also just click on the field and change number directly. I want to keep the style of the original sheet as much as possible.

>     * [Improvement] For Character Sheet, make it a vertical single column of section starting from name, XP & Gold, Resources, Masteries, Perks, and then Notes. For the Perk section, do not put the check boxes and text in the same row. One row for check boxes and one row for text.

Same as party sheet, don't want to touch too much on the sheet itself. Already make the boxes much bigger to be accessible on mobile. Putting it in multiple lines will create even more scrolling which I don't like. 

> 
>     * [Improvement] For the items part of the Character Sheet, put the drop down on the left-most followed by the number input box with an "ADD" button closely tied to the input box. I don't think the left and right arrows are helpful (when potentially navigating hundreds of items sorted in mysteriously numbered order). Adding a 'Available Items' button to open up an item list/index would be a better option.

>     * Thank you for the switching forms function!! It is so amazing and amazingly useless (gameplay-wise), but I loved it so much. I got a funny second name for my Geminate and it shows!!!! It is such attentions to details made me in love with you.

For that one, you have to thank other user, which requested this in #210.

> Thank you for this amazing web app!! LOVE YOU!!!!!!!!!! I have tried x-haven but I think GS is much more appealing visually and it is an absolute priority for me! Glad we have so many amazing, humble and talented people willing to pick up the legacy of GHH. I would give you a huge kiss if I can. May you loot all the coins in the loot deck.

Haha thanks.

**Comment by Lurkars** [comment]
Already handled some small stuff in [v0.67.9](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.67.9). Happy about more feedback.

**Comment by eaganlai** [comment]
For the Feedback, the message was probably too long since I was trying to send this there.

My mobile is Nokia X10, it does have a bigger screen size than most.

For the Avatar one, I think I didn't make it clear. I hope to see the full picture of the monster, but not just a enlarged circle.

Thanks!! I will check the changes!!

**Comment by Lurkars** [comment]
> For the Avatar one, I think I didn't make it clear. I hope to see the full picture of the monster, but not just a enlarged circle.

Haha okay, for many monsters, I don't have other pictures.

**Comment by Lurkars** [comment]
> > For the Avatar one, I think I didn't make it clear. I hope to see the full picture of the monster, but not just a enlarged circle.
> 
> Haha okay, for many monsters, I don't have other pictures.

Added monster artwork I could extract from TTS in [v0.68.4](https://github.com/Lurkars/gloomhavensecretariat/releases/tag/v0.68.4) 

Hope that's what you meant.

**Comment by eaganlai** [comment]
I have another run with my friends last weekend! I love all the changes!

I know it is my fault. With the enlarged artwork added, now it becomes quite hard to drag the order around. Also it seems the adding items on character sheets look entirely different on Android(Firefox) and iOS(Safari), I haven't confirm this with more iPhones yet.

**Comment by Lurkars** [comment]
Weird, can you provide screenshots? I added setting to disable "portrait mode optimization" caused by changes here. Also possible that versions do not match? 

Anyways, regarding the artwork and drag'n'drop, I will change artwork popping up only after release the long press, so you have "endless" time to decide for dragging or artwork, hope that works out and fixes that issue.

**Comment by mentonin** [comment]
> I want to keep the style of the original sheet as much as possible.

Would you consider an option to change between "classic" sheets and a "digital" one? A design like the Gloomhaven Companion's might be preferable for many people.

> For maintaining reasons I already stated out somewhere that I don't want to change too much stuff between editions

The loot indicator could be toggled based on Loot Deck presence, which is already different between editions. It also makes much more sense to have a loot card indicator than a coin indicator when you have loot



>>    * [Improvement] When setting Initiative, simply put a wide bar button saying "Long Rest" under zero is more intuitive than a 'history' icon.
> 
> I am unsure about this, because there are many thinks that may not be intuitive and you just need to learn. If you once knew it, it just looks cleaner and easier then always having a big text button on this view. But we can discuss on that.

I also think the big text button is unclean. I think a skull and a "Zzz" button next to 0 would be best, for exhausting and long resting. Would also mean changing the in-round exhaust button to a skull, which I believe makes more sense

**Comment by Lurkars** [comment]
> Would you consider an option to change between "classic" sheets and a "digital" one? A design like the Gloomhaven Companion's might be preferable for many people.

Yeah this sounds like a good idea. Biggest issue for me: UX. I am really bad and if I design a digital version myself, everyone will still prefer the classical sheet :sweat_smile: . So if anyone open for proposals, go for it.

> The loot indicator could be toggled based on Loot Deck presence, which is already different between editions. It also makes much more sense to have a loot card indicator than a coin indicator when you have loot

Yes, for FH you're right, would make more sense to track the coin loot as other resources. I will think about it.

> I also think the big text button is unclean. I think a skull and a "Zzz" button next to 0 would be best, for exhausting and long resting. Would also mean changing the in-round exhaust button to a skull, which I believe makes more sense

Haha, this was just request in this ticket and I did this. So you're more asking about reverting the change. I don't care that much about that small button, so I am fine to keep it as it is for better readability because there is no official iconography for such things and I defined those myself, it may be easier.

**Comment by mentonin** [comment]
> Biggest issue for me: UX

The app I mentioned, Gloomhaven Companion, has a decent enough interface. I don't think it is open, but should be feasible to copy.

> So you're more asking about reverting the change.

I was just agreeing with you, since you said you were unsure about it. It is not a big deal, indeed.

**Comment by Lurkars** [comment]
Since most of the stuff is implemented, I will close this issue and create a separate issue for creating alternate sheets.
